go/tasks/plugins/k8s/spark/spark.go

```diff
@@ -407,8 +407,9 @@ func getEventInfoForSpark(sj *sparkOp.SparkApplication) (*pluginsCore.TaskInfo,
 			})
 		}
 	} else if sj.Status.AppState.State == sparkOp.RunningState && sj.Status.DriverInfo.WebUIIngressAddress != "" {
-		// Append https as the operator doesn't currently.
-		customInfoMap[sparkDriverUI] = fmt.Sprintf("https://%s", sj.Status.DriverInfo.WebUIIngressAddress)
+		// Older versions of spark-operator does not append http:// but newer versions do.
+		uri := strings.TrimPrefix(sj.Status.DriverInfo.WebUIIngressAddress, "http://")
+		customInfoMap[sparkDriverUI] = fmt.Sprintf("https://%s", uri)
```

Comment:
We shouldn't assume SSL all the time.
```
uri := sj.Status.DriverInfo.WebUIIngressAddress
if !strings.HasPrefix("https://") && !strings.HasPrefix("http://") {
    uri = fmt.Sprintf("https://%s", uri)
}

customInfoMap[sparkDriverUI] = uri
```

Reply:
I agree, so we should just pass the `WebUIIngressAddress` value along?

Reply:
@mucahitkantepe yes, and then if it does not contain `https` or `http` we can prepend with `https` as Haytham suggests.